src/Range/AbstractRange.php

```diff
@@ -122,4 +124,49 @@ public function containsRange(RangeInterface $range)
 
         return $result;
     }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @see \IPLib\Range\RangeInterface::split()
+     */
+    public function split($networkPrefix)
+    {
+        if ($this instanceof Single){
+            return array(clone $this);
+        }
+
+        $networkPrefix = (int) $networkPrefix;
+        $myNetworkPrefix = $this->getNetworkPrefix();
+        if ($networkPrefix === $myNetworkPrefix) {
+            return array(
+                new Subnet($this->getStartAddress(), $this->getEndAddress(), $networkPrefix),
```

Comment:
@mlocati
I also suggest making this line `static` so the return type is based on the class.

Would you be okay with this change?

Reply:
We have two options: thr split method always return Subnet instances, or it always returns instances of the range that's running the split method.
And I'd prefer this second option.
So: yes!